firedrake/tsfc_interface.py

```diff
@@ -156,7 +154,7 @@ def _compile_form_hashkey(form, name, parameters=None, split=True, interface=Non
     cachedir=_cachedir
 )
 @PETSc.Log.EventDecorator()
-def compile_form(form, name, parameters=None, split=True, interface=None, diagonal=False):
+def compile_form(form, name, parameters=None, split=True, dont_split=None, diagonal=False):
```

Comment:
Can the default for `dont_split` be an empty tuple? It makes it a little easier to understand.